src/charm.py

```diff
@@ -44,6 +44,10 @@ def _detect_java_application(self) -> ExecutableJarApplication:
                 )
             jar_file = jar_files[0]
             return ExecutableJarApplication(executable_jar_path=f"/app/{jar_file}")
+        if container.exists(
```

Comment:
I'd probably also fallback to the "buildpack" case if "/app" exists but no jar is found in the folder

Reply:
I moved the buildpack detection before executable jar detection. Now the buildpack image can be detected even if it contains the `/app` directory.